AppCenter/AppCenter/Internals/Channel/MSChannelUnitDefault.m

```diff
@@ -223,106 +348,8 @@ - (void)flushQueue {
-                                                          [delegate channel:self didSucceedSendingLog:aLog];
-                                                        }
-                                                      }];
-
-                            // Remove from pending logs and storage.
-                            [self.pendingBatchIds removeObject:ingestionBatchId];
-                            [self.storage deleteLogsWithBatchId:ingestionBatchId groupId:self.configuration.groupId];
-
-                            // Try to flush again if batch queue is not full anymore.
-                            if (self.pendingBatchQueueFull && self.pendingBatchIds.count < self.configuration.pendingBatchesLimit) {
-                              self.pendingBatchQueueFull = NO;
-                              if (self.availableBatchFromStorage) {
-                                [self flushQueue];
-                              }
-                            }
-                          }
-
-                          // Failure.
-                          else {
-                            MSLogError([MSAppCenter logTag], @"Log(s) sent with failure, batch Id:%@, status code:%tu", ingestionBatchId,
-                                       response.statusCode);
-
-                            // Notify delegates.
-                            [self enumerateDelegatesForSelector:@selector(channel:didFailSendingLog:withError:)
-                                                      withBlock:^(id<MSChannelDelegate> delegate) {
-                                                        for (id<MSLog> aLog in logArray) {
-                                                          [delegate channel:self didFailSendingLog:aLog withError:error];
-                                                        }
-                                                      }];
-
-                            // Remove from pending logs.
-                            [self.pendingBatchIds removeObject:ingestionBatchId];
-                            [self.storage deleteLogsWithBatchId:ingestionBatchId groupId:self.configuration.groupId];
-
-                            // Update pending batch queue state.
-                            if (self.pendingBatchQueueFull && self.pendingBatchIds.count < self.configuration.pendingBatchesLimit) {
-                              self.pendingBatchQueueFull = NO;
-                            }
-                          }
-                        } else
-                          MSLogWarning([MSAppCenter logTag], @"Batch Id %@ not expected, ignore.", ingestionBatchId);
-                      });
-                    }];
-          }
-        }];
-
-  // Flush again if there is another batch to send.
-  if (self.availableBatchFromStorage && !self.pendingBatchQueueFull) {
-    [self flushQueue];
-  }
+  NSMutableArray<MSAuthTokenValidityInfo *> *tokenArray = [MSAuthTokenContext sharedInstance].authTokenValidityArray;
```

Comment:
Can it be `[[MSAuthTokenContext sharedInstance] authTokenValidityArray]`? There is a custom getter, it can be accessed like that.